Rakefile

```diff
@@ -425,3 +425,288 @@ end
+            validate_not_null_or_empty: true
+          }
+        end
+
+        help_text[:flags].reject { |o| o =~ /verbose|debug|help/ }.map do |option|
+          ruby_param = parser.top.long[option]
+          pwsh_name = option.gsub("-", "")
+
+          next if pwsh_name == 'version'
+
+          pwsh_param = {
+            name:       pwsh_name,
+            type:       'string',
+            switch:     false,
+            mandatory:  false,
+            help_msg:   ruby_param.desc.map { |d| d.gsub("<", "&lt;") }.join("\n"),
+            ruby_short: ruby_param.short.first,
+            ruby_long:  ruby_param.long.first,
+            ruby_arg:   ruby_param.arg,
+            ruby_orig:  option
+          }
+
+          case ruby_param.class.to_s
+          when 'OptionParser::Switch::RequiredArgument'
+            # this isn't quite the same thing as a mandatory pwsh parameter
+            # pwsh_param[:mandatory] = true
+          when 'OptionParser::Switch::OptionalArgument'
+            pwsh_param[:mandatory] = false
+          when 'OptionParser::Switch::NoArgument'
+            pwsh_param[:mandatory] = false
+            pwsh_param[:switch] = true
+            pwsh_param[:type] = 'switch'
+          end
+
+          # Only one of --targets , --rerun , or --query can be used
+          # Only one of --configfile or --boltdir can be used
+          case pwsh_name
+          when 'user'
+            pwsh_param[:validate_not_null_or_empty] = true
+          when 'password'
+            pwsh_param[:validate_not_null_or_empty] = true
+          when 'privatekey'
+            pwsh_param[:validate_not_null_or_empty] = true
+          when 'concurrency'
+            pwsh_param[:validate_not_null_or_empty] = true
+          when 'compileconcurrency'
+            pwsh_param[:validate_not_null_or_empty] = true
+          when 'connecttimeout'
+            pwsh_param[:validate_not_null_or_empty] = true
+          when 'modulepath'
+            pwsh_param[:validate_not_null_or_empty] = true
+          when 'transport'
+            pwsh_param[:validate_set] = %w[winrm ssh pcp]
+          when 'targets'
+            pwsh_param[:validate_not_null_or_empty] = true
+          when 'query'
+            pwsh_param[:validate_not_null_or_empty] = true
+          when 'rerun'
+            pwsh_param[:validate_not_null_or_empty] = true
+          when 'configfile'
+            pwsh_param[:validate_not_null_or_empty] = true
+          when 'boltdir'
+            pwsh_param[:validate_not_null_or_empty] = true
+          when 'execute'
+            pwsh_param[:parameter_set] = 'execute'
+            pwsh_param[:mandatory] = true
+            pwsh_param[:position] = 0
+          when 'params'
+            pwsh_param[:position] = 1
+            pwsh_param[:type] = nil
+          when 'modules'
+            pwsh_param[:type] = nil
+          when 'format'
+            pwsh_param[:validate_set] = %w[human json rainbow]
```

Comment:
:rainbow:

Reply:
Oh! Actually, we don't support rainbow on Windows. We specifically check the OS, not that we're in powershell, which may be a mistake since I think this won't work in powershell not just on Windows...

Reply:
Displaying ansi escape codes is a limitation of the terminal, both Windows and PowerShell 6/7 support them. We can investigate adding that baack if we detect the user is using Windows Terminal, or one of the Windows 10 conhosts in another ticket.

Reply:
#1983 will add support for rainbow on windows, so I vote for keeping the enum with the full set here